log4j-api-test/src/test/java/org/apache/logging/log4j/message/ParameterFormatterTest.java

```diff
@@ -17,21 +17,27 @@
 package org.apache.logging.log4j.message;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import org.apache.logging.log4j.message.ParameterFormatter.MessagePatternAnalysis;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.apache.logging.log4j.test.junit.UsingStatusLoggerMock;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.CsvSource;
 import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
 
 /**
  * Tests {@link ParameterFormatter}.
  */
-public class ParameterFormatterTest {
+@UsingStatusLoggerMock
+class ParameterFormatterTest {
```

Comment:
I am not a big fan of mocking the entire `StatusLogger`: this prevents the test to run in parallel with other tests.

I prefer the [`@UsingStatusListener`](https://github.com/apache/logging-log4j2/blob/2.x/log4j-api-test/src/main/java/org/apache/logging/log4j/test/junit/UsingStatusListener.java) extension to register a per-test status listener.

Of course neither of us is impartial in the choice of JUnit extension to use. ;-)

Reply:
Piotr, if I would have been a programmer who also happens to be a serial killer, you would not want me to see what you have done in `UsingStatusListener` et al. using `ThreadLocal`s. :sunglasses: I find it a liability with several fragile assumptions just to make tests needing `StatusLogger` work in parallel. I have so much dislike towards such design choices, but I will spare that for another time.

Just to show how much I like you, I have replaced `@UsingStatusLoggerMock` with `@UsingStatusListener` in 186c580d2cda1f301e7b5eab14ded8a15e5b54f5 as you requested. :stuck_out_tongue_closed_eyes: